[QUESTION] Extra header files needed to Common.c when adding createLwsIotCredentialProvider?

After adding
createLwsIotCredentialProvider to Common.c using a Mac
I get the following bunch of linking errors.
So any ideas what's going on? Do I need to add extra header files from libwebsockts to solve this?
```
Undefined symbols for architecture x86_64:
  "_lws_add_http_header_by_name", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_callback_on_writable", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_cancel_service", referenced from:
      _blockingLwsCall in libkvsCommonLws.a(LwsCall.c.o)
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_client_connect_via_info", referenced from:
      _blockingLwsCall in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_client_http_body_pending", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_context_destroy", referenced from:
      _blockingLwsCall in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_create_context", referenced from:
      _blockingLwsCall in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_get_context", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_get_opaque_user_data", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_hdr_custom_copy", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_http_client_http_response", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_http_client_read", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_service", referenced from:
      _blockingLwsCall in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_set_log_level", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lws_write", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
  "_lwsl_hexdump_level", referenced from:
      _lwsIotCallbackRoutine in libkvsCommonLws.a(LwsCall.c.o)
ld: symbol(s) not found for architecture x86_64

```

The one above is from Mac linking. The following is the error from arm cross building
```
/usr/local/linaro-armv7ahf-2015.11-gcc5.2/bin/../lib/gcc/arm-linux-gnueabihf/5.2.1/../../../../arm-linux-gnueabihf/bin/ld: libkvsWebrtcSignalingClient.so: undefined reference to symbol 'lws_http_client_read'
/root/host/VIGIL_Edge_Cloud/kvs-webrtc/open-source/lib/libwebsockets.so.15: error adding symbols: DSO missing from command line
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -270,14 +270,15 @@ add_library(kvsWebrtcSignalingClient ${LINKAGE} ${WEBRTC_SIGNALING_CLIENT_SOURCE
 target_link_libraries(
   kvsWebrtcSignalingClient
   PRIVATE kvspicUtils
-          kvspicState
-          ${CMAKE_THREAD_LIBS_INIT}
-          ${LIBWEBSOCKETS_LIBRARIES}
-          ${EXTRA_DEPS}
-          ${OPENSSL_SSL_LIBRARY}
-          ${OPENSSL_CRYPTO_LIBRARY}
-          ${MBEDTLS_LIBRARIES}
-  PUBLIC kvsCommonLws)
+         kvspicState
+         ${CMAKE_THREAD_LIBS_INIT}
+         ${EXTRA_DEPS}
+         ${OPENSSL_SSL_LIBRARY}
+         ${OPENSSL_CRYPTO_LIBRARY}
+         ${MBEDTLS_LIBRARIES}
+  PUBLIC
+        ${LIBWEBSOCKETS_LIBRARIES} 
+        kvsCommonLws)
 
 if (WIN32)
   target_link_libraries(kvsWebrtcClient PRIVATE "Ws2_32" "iphlpapi")

